datafusion/sqllogictest/test_files/insert_to_external.slt

```diff
@@ -456,13 +456,16 @@ explain insert into table_without_values select c1 from aggregate_test_100 order
 ----
 logical_plan
 01)Dml: op=[Insert Into] table=[table_without_values]
-02)--Projection: aggregate_test_100.c1 AS c1
+02)--Projection: CAST(aggregate_test_100.c1 AS Utf8View) AS c1
```

Comment:
This is also expected,  for the sql way insert into, we will automatically cast the type to match the table schema, see details:

The code here:
https://github.com/apache/datafusion/blob/2011f5252c7a738854f402c70a6636aaa408aaec/datafusion/sql/src/statement.rs#L1958

Reply:
Created a follow-up ticket for it, we'd better to support cast also for insert into api way consistent with insert into sql.
https://github.com/apache/datafusion/issues/15015